hyperspy/signal.py

```diff
@@ -3169,6 +3170,115 @@ def integrate_simpson(self, axis, out=None):
+            im_fft = self._deepcopy_with_new_data(np.fft.fftshift(
+                np.fft.fftn(self.data, axes=axes, **kwargs), axes=axes))
+        else:
+            im_fft = self._deepcopy_with_new_data(np.fft.fftn(self.data, axes=axes, **kwargs))
+
+        im_fft.change_dtype("complex")
+        im_fft.metadata.General.title = 'FFT of {}'.format(im_fft.metadata.General.title)
+        im_fft.metadata.Signal.signal_type = 'FFT of {}'.format(im_fft.metadata.Signal.signal_type)
+        ureg = UnitRegistry()
+        for axis in im_fft.axes_manager.signal_axes:
+            axis.scale = 1. / axis.size / axis.scale
+            try:
+                units = ureg.parse_expression(str(axis.units))**(-1)
+                axis.units = '{:~}'.format(units.units)
+            except UndefinedUnitError:
+                _logger.warning('Units are not set or cannot be recognized')
+            if shifted:
+                axis.offset = -axis.high_value / 2.
+        return im_fft
+
+    def ifft(self, shifted=False, **kwargs):
+        """
+        Compute the inverse discrete Fourier Transform.
+
+        This function computes real part of the inverse of the discrete
+        Fourier Transform over the signal axes by
+        means of the Fast Fourier Transform (FFT).
+
+        Parameters
+        ----------
+        shifted : bool, optional
+            If True, the origin of FFT will be shifted in the centre (Default: False).
+
+        Return
+        ------
+        s : Signal
+
+        Examples
+        --------
+        >>> im = hs.signals.Signal2D(scipy.misc.lena())
+        >>> imfft = im.fft()
+        >>> imfft.ifft()
+        <Image, title: iFFT of FFT of , dimensions: (|512, 512)>
+
+        Notes
+        -----
+        For further information see the documentation of numpy.fft.ifftn
+
+        """
+
+        if self.axes_manager.signal_dimension == 0:
+            raise AttributeError("Signal dimension must be at least one.")
+        ax = self.axes_manager
+        axes = ax.signal_indices_in_array
+
+        if shifted:
+            im_ifft = self._deepcopy_with_new_data(np.fft.ifftn(np.fft.ifftshift(
+                self.data, axes=axes), axes=axes, **kwargs))
+        else:
+            im_ifft = self._deepcopy_with_new_data(np.fft.ifftn(
+                self.data, axes=axes, **kwargs))
+
+        im_ifft.metadata.General.title = 'iFFT of {}'.format(im_ifft.metadata.General.title)
+        im_ifft.metadata.Signal.signal_type = 'iFFT of {}'.format(im_ifft.metadata.Signal.signal_type)
+        im_ifft = im_ifft.real
+        ureg = UnitRegistry()
+        for axis in im_ifft.axes_manager.signal_axes:
+            axis.scale = 1. / axis.size / axis.scale
+            try:
+                units = ureg.parse_expression(str(axis.units)) ** (-1)
+                axis.units = '{:~}'.format(units.units)
+            except UndefinedUnitError:
+                _logger.warning('Units are not set or cannot be recognized')
```

Comment:
I actually think that keeping original units is better approach... For example if someones units are nanometers written with typo, doing fft and than ifft in that case will result in the same units rather than loosing them completely. I'm also not sure how good is units coverage of pint package.

Reply:
Pint is quite good, it will deal with units (and potential encoding related issue, that we can get with um, angstrom...) much better than we will do!
If there is a typo, the warning will first point it out and removing the unit may force the user to correct it (where the actual error is)! ;) 

In this specific case, the benefit of using pint are quite marginal thought... so maybe it is not worth it...